src/pke/lib/scheme/bfvrns/bfvrns-parametergeneration.cpp

```diff
@@ -83,14 +83,15 @@ bool ParameterGenerationBFVRNS::ParamsGenBFVRNS(std::shared_ptr<CryptoParameters
 
     DistributionType distType;
 
+    uint32_t thresholdParties = cryptoParamsBFVRNS->GetThresholdNumOfParties();
     // supports both discrete Gaussian (GAUSSIAN) and ternary uniform distribution
     // (UNIFORM_TERNARY) cases
     if (cryptoParamsBFVRNS->GetSecretKeyDist() == GAUSSIAN) {
-        Bkey     = sigma * sqrt(alpha);
+        Bkey     = sqrt(thresholdParties) * sigma * sqrt(alpha);
         distType = HEStd_error;
     }
     else {
-        Bkey     = 1;
+        Bkey     = thresholdParties * 1;
```

Comment:
you don't need this multiplication by 1, an assignment would suffice. "thresholdParties * 1;" can be tranformed to a comment